### ISSUE TITLE
The result is inconsistent with Java

### ISSUE THREAD

**Issue by sunxiongwei** [mvel/mvel#280]
public static void main(String[] args) {
        int i = 6.7 - 100 > 39.6 ? 5 == 5 ? 4 + 5 : 6 - 1 : !(100 % 3 - 39.0 < 27) ? 8 * 2 - 199 : 100 % 3;
        System.out.println(i);
        Object o = MVEL.eval("6.7 - 100 > 39.6 ? 5 == 5 ? 4 + 5 : 6 - 1 : !(100 % 3 - 39.0 < 27) ? 8 * 2 - 199 : 100 % 3");
        System.out.println(o);
    }

java return 1;
mvel return 5;

why? a bug?

**Comment by MalcolmOdd** [comment]
Hi @sunxiongwei, you make your example unnecessarily complicated, is this a puzzle of the day thing ? :)
Here is a pair of test cases that reveal that the operator priority between the ternary operator ( ?: ) and the equal comparison ( == ) differs between Java and mvel - the second test adds parenthesis to force mvel to use the same priority as Java so it succeeds:
```
	public void testTernaryOperatorPriorityFails () {
		int javaResult = false ? 5 == 5 ? 9 : 5 : 1;
		Integer mvelResult = MVEL.eval("false ? 5 == 5 ? 9 : 5 : 1", Integer.class);
	    Assert.assertEquals(javaResult, mvelResult.intValue());
	}

	public void testTernaryOperatorPriorityPasses () {
		int javaResult = false ? 5 == 5 ? 9 : 5 : 1;
		Integer mvelResult = MVEL.eval("false ? (5 == 5 ? 9 : 5) : 1", Integer.class);
	    Assert.assertEquals(javaResult, mvelResult.intValue());
	}

```

**Comment by sunxiongwei** [comment]
> Hi @sunxiongwei, you make your example unnecessarily complicated, is this a puzzle of the day thing ? :) Here is a pair of test cases that reveal that the operator priority between the ternary operator ( ?: ) and the equal comparison ( == ) differs between Java and mvel - the second test adds parenthesis to force mvel to use the same priority as Java so it succeeds:
> 
> ```
> 	public void testTernaryOperatorPriorityFails () {
> 		int javaResult = false ? 5 == 5 ? 9 : 5 : 1;
> 		Integer mvelResult = MVEL.eval("false ? 5 == 5 ? 9 : 5 : 1", Integer.class);
> 	    Assert.assertEquals(javaResult, mvelResult.intValue());
> 	}
> 
> 	public void testTernaryOperatorPriorityPasses () {
> 		int javaResult = false ? 5 == 5 ? 9 : 5 : 1;
> 		Integer mvelResult = MVEL.eval("false ? (5 == 5 ? 9 : 5) : 1", Integer.class);
> 	    Assert.assertEquals(javaResult, mvelResult.intValue());
> 	}
> ```

Thank you. Got it！

**Comment by MalcolmOdd** [comment]
Hi @sunxiongwei , I saw you closed this issue but an operator priority inversion is actually an issue.
Furthermore, I found that the issue is not between the ternary operator and the equal comparison but rather between ternary operators. it can be reproduced with the expression "false ? true ? 9 : 5 : 1" which doesn't have the comparison.

The cause of this issue is at the following lines: https://github.com/mvel/mvel/blob/a2f8d2e71b7a315de106c65f1ccc15d15853a1e1/src/main/java/org/mvel2/MVELRuntime.java#L103 and https://github.com/mvel/mvel/blob/a2f8d2e71b7a315de106c65f1ccc15d15853a1e1/src/main/java/org/mvel2/MVELInterpretedRuntime.java#L247
(The code is different depending on whether you call compileExpression+executeExpression or eval and the bug is present in both cases).

Adding parenthesis is a workaround.
